lib/AST/Mangle.cpp

```diff
@@ -1603,7 +1603,12 @@ void Mangler::mangleEntity(const ValueDecl *decl,
       return mangleAccessorEntity(accessorKind, func->getAddressorKind(),
                                   func->getAccessorStorageDecl());
   }
-  
+
+  // Avoid mangling nameless entity. This may happen in erroneous code as code
+  // completion.
+  if (!decl->hasName())
```

Comment:
Why does it not have a name? Would it be better to ensure a name is always set? Or maybe we shouldn't be trying to mangle the entity in the first place? Or perhaps we should fix the mangler to just not allow this?